transformers/modeling_utils.py

```diff
@@ -245,17 +245,20 @@ def save_pretrained(self, save_directory, xla_device=False):
         # Only save the model itself if we are using distributed training
         model_to_save = self.module if hasattr(self, 'module') else self
 
-        # Save configuration file
-        model_to_save.config.save_pretrained(save_directory)
-
         # If we save using the predefined names, we can load using `from_pretrained`
         output_model_file = os.path.join(save_directory, WEIGHTS_NAME)
+
+        # Save configuration file
         if xla_device:
-            # Saving for each process is save since output_dir is proc# namespaced.
             import torch_xla.core.xla_model as xm
-            xm.save(model_to_save.state_dict(), output_model_file, master_only=False)
+            if xm.is_master_ordinal():
+                model_to_save.config.save_pretrained(save_directory)
+            # xm.save takes care of saving only from master
+            xm.save(model_to_save.state_dict(), output_model_file)
```

Comment:
idea for the future; you could use the checkpoint tagger later, if they mark which chpt is the best etc.

Reply:
As far as I know, they don't have such tagger but let me see if they do. Good idea we could even upstream if they don't. Thanks.